### PR TITLE
Ivy layer bugfixes for #12838

### PR DESCRIPTION
**Pull request by Caribpa**
Hi!

These are some bugfixes for #12838

Thanks!

### CODE REVIEW

File: layers/+completion/ivy/funcs.el [hunk 1]
```diff
@@ -254,27 +254,30 @@ that directory."
                    (buffer-substring-no-properties
                     (region-beginning) (region-end))
                  (thing-at-point 'symbol t))))
-    (counsel-git-grep nil input)))
+    (counsel-git-grep input)))
```
Comment (xuhdev) [inline layers/+completion/ivy/funcs.el]: I'm not sure whether this function is useful at all. Thing at point can be better guessed by running `counsel-git-grep` followed by `M-n`.

Comment (duianto) [reply layers/+completion/ivy/funcs.el]: `M-n` which calls `ivy-previous-history-element`, only cycles backwards through the prompts history. If one hasn't manually typed in the symbol under the point then it won't be listed.

This PR fixes the issue where `SPC s g P` behaved the same as `SPC s g p` and didn't insert the symbol under the point.

**Comment by duianto** [comment]
Thank you for contributing to Spacemacs. It's safe to delete your branch.
The PRs changes have been cherry picked into the `develop` branch.
https://github.com/syl20bnr/spacemacs/commit/cb99bf9f390abbdcd4f2760297037fa1752bf9c6
https://github.com/syl20bnr/spacemacs/commit/0b0203e03f7bb67c7ce10fd08677f9d4a528d7a3

File: layers/+completion/ivy/funcs.el (hunk 1 above)
Comment (xuhdev) [reply layers/+completion/ivy/funcs.el]: @duianto `M-n` calls the next (not previous) element in history. When the next element does not exist, it tries to predict what the user wants to input, in this case, what the thing at point is. See https://oremacs.com/2016/02/15/counsel-ffap/

Comment (duianto) [reply layers/+completion/ivy/funcs.el]: Your right, `M-n` does pick up the symbol under the cursor.
I did not know that. Thanks for the explanation.

I'm used to navigating backwards through the history at prompts, therefore I assumed that you meant `M-p`, but I still managed to type the correct key binding: `M-n` 😄 

Still, `SPC s g P` was broken, but it's fixed now.

File: layers/+completion/ivy/funcs.el [hunk 2]
```diff
@@ -254,27 +254,30 @@ that directory."
...
   (spacemacs/counsel-search dotspacemacs-search-tools
                             nil spacemacs-docs-directory))
 
-(defun spacemacs//counsel-occur ()
+(defun spacemacs//counsel-occur (&optional candidates)
   "Generate a custom occur buffer for `counsel-git-grep'."
   (ivy-occur-grep-mode)
   (setq default-directory (ivy-state-directory ivy-last))
-  (let ((cands ivy--old-cands))
+  (let ((cands (if (boundp 'candidates)
```
Comment (yuhan0) [inline layers/+completion/ivy/funcs.el]: I see this PR has already been merged but this should just be `(or candidates ivy--old-cands)`, the symbol itself is always bound in this context.

Comment (duianto) [reply layers/+completion/ivy/funcs.el]: Thanks for the suggestion. Fixed: https://github.com/syl20bnr/spacemacs/commit/b837c1d39705b9e03bd9922da249e4c154023a92

**Pull request closed without merging**